x/poolmanager/router_test.go

```diff
@@ -1530,3 +1588,419 @@ func (suite *KeeperTestSuite) setupPools(poolType types.PoolType, poolDefaultSwa
 		return
 	}
 }
+
+func (suite *KeeperTestSuite) TestSplitRouteExactAmountIn() {
+	var (
+		defaultSingleRouteOneHop = []types.SwapAmountInSplitRoute{
+			{
+				Pools: []types.SwapAmountInRoute{
+					{
+						PoolId:        fooBarPoolId,
+						TokenOutDenom: bar,
+					},
+				},
+				TokenInAmount: twentyFiveBaseUnitsAmount,
+			},
+		}
+
+		defaultTwoHopRoutes = []types.SwapAmountInRoute{
+			{
+				PoolId:        fooBarPoolId,
+				TokenOutDenom: bar,
+			},
+			{
+				PoolId:        barBazPoolId,
+				TokenOutDenom: baz,
+			},
+		}
+
+		defaultSingleRouteTwoHops = types.SwapAmountInSplitRoute{
+			Pools:         defaultTwoHopRoutes,
+			TokenInAmount: twentyFiveBaseUnitsAmount,
+		}
+
+		defaultSingleRouteThreeHops = types.SwapAmountInSplitRoute{
+			Pools: []types.SwapAmountInRoute{
+				{
+					PoolId:        fooBarPoolId,
+					TokenOutDenom: bar,
+				},
+				{
+					PoolId:        barUosmoPoolId,
+					TokenOutDenom: uosmo,
+				},
+				{
+					PoolId:        bazUosmoPoolId,
+					TokenOutDenom: baz,
+				},
+			},
+			TokenInAmount: sdk.NewInt(twentyFiveBaseUnitsAmount.Int64() * 3),
+		}
+
+		priceImpactThreshold = sdk.NewInt(97866545)
```

Comment:
Can you add a comment explaining where this number comes from? It made sense after I read through the tests, but would be nice to have the context prior to then